test/test_multithread_stress.c

```diff
@@ -222,7 +224,10 @@ int main(int argc, char **argv)
 	printf("Test Interval  :\t%d\n", test_interval);
 	printf("Test Iterations:\t%d\n",test_iterations);
 
-	generate_data_buffer(data_buf);
+	if (generate_data_buffer(data_buf) != 0) {
+		free_data_buffer(data_buf);
```

Comment:
Is this 100% safe? If the OOM happens very early in `generate_data_buffer`, it will exit without initializing some elements in `buffer`, not even with `NULL`. So `free_data_buffer` may call `free` on uninitialized pointers later. The man page for `free` does not make it really clear what happens in this case.

Reply:
Actually this is just fixed if the `data_buf` is initialized with `NULL`.